specifyweb/frontend/js_src/lib/components/Molecules/ResourceLink.tsx

```diff
@@ -39,9 +40,12 @@ export function ResourceLink<COMPONENT extends typeof Link['Icon']>({
     >,
     'dialog' | 'onAdd' | 'onClose' | 'onSaved'
   >;
+  readonly autoClose?: boolean;
 }): JSX.Element {
   const [isOpen, _, handleClose, handleToggle] = useBooleanState();
-  React.useEffect(handleClose, [resource]);
+  React.useEffect(() => {
+    if (autoClose === undefined || autoClose) handleClose();
+  }, [resource]);
```

Comment:
your version is fine, but just showing that this variant is equivalent:
```suggestion
  React.useEffect(autoClose ? handleClose : f.void, [resource]);
```

or, if you want to placate the eslint rule that complains about the dependency array, could do this:

```suggestion
  React.useEffect(() => {
    // Close the dialog when resource changes
    void resource;
    if (autoClose === undefined || autoClose) handleClose();
  }, [handleClose, resource]);
```

(i.e we are needlessly referring the the `resource`, thus ESLint stops complaining about it being needlessly specified in the dependency array)